src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,81 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, {
+  useCallback, useContext, useEffect, useState,
+} from 'react';
+import { FilterParams } from '../../types/filterParams';
+import { TodosContext } from '../Store/Store';
+import { getFilteredTodos } from '../../services/getfilteredTodos';
+
+export const TodoFilter = React.memo(() => {
+  const {
+    todos,
+    setFilteredTodos,
+  } = useContext(TodosContext);
+
+  const [title, setTitle] = useState('');
+  const [filter, setFilter] = useState(FilterParams.All);
+
+  const filteredTodos = useCallback(() => {
+    return getFilteredTodos(todos, filter, title);
+  }, [todos, filter, title]);
+
+  useEffect(() => {
+    setFilteredTodos(filteredTodos());
+  }, [filter, title]);
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.toLowerCase());
```

Comment:
do not transform input value to lowerCase. Now I'm not able to type uppercased text in the input. transform to one case in your getFilteredTodos fn.